File: pkg/model/venue.go
package model

import (
	"time"

	"gorm.io/gorm"
)

type Venue struct {
	gorm.Model
	Name        string     `json:"name"`
	Description string     `json:"description"`
	Address     string     `json:"address"`
	ZipCode     int        `json:"zip"`
	City        string     `json:"city"`
	Slots       []Timeslot `json:"slots"`
	User        User       `json:"-"`
	UserID      uint       `json:"userID"`
	Contact     string     `json:"contact"`
}

type Timeslot struct {
	gorm.Model
	VenueID  uint      `json:"venueID"`
	Time     time.Time `json:"time"`
	Artist   Artist    `json:"artist"`
	ArtistID uint      `json:"artistID"`
	Pay      int       `json:"pay"`
	Private  bool      `json:"private"`
	Duration int       `json:"duration"`
}

File: ui/src/app/models/venue.ts
import { Artist } from "./artist"

export interface Venue {
  ID: number
  CreatedAt?: string
  name: string
  description: string
  slots: Timeslot[]
  contact: string
  address: string
  zip: number
  city: string
  phone?: string
}

export interface Timeslot {
  ID: number
  time: Date
  artist?: Artist
  artistID: number
  venueID: number
  pay: number
  private: boolean
  duration: number
}
